app/models/ui/UIComponent.scala

```diff
@@ -58,6 +58,14 @@ case class Text(english: Seq[TextItem], welsh: Seq[TextItem]) {
   def asString(implicit lang: Lang): String = toWords.mkString(" ")
   override def toString: String = s"[${english.map(t => t.toString).mkString("")}:${welsh.map(t => t.toString).mkString("")}]"
   def +(other: Text): Text = Text(english ++ other.english, welsh ++ other.welsh)
+  lazy val isBold: Boolean = english.length == 1 && (english.head match {
+    case w: Words => w.bold
+    case _ => false
+  })
+  lazy val isNumeric: Boolean = english.length == 1 && (english.head match {
```

Comment:
The function name isNumeric is a bit misleading. You are really checking a label value through a label reference. Perhaps isNumericLabelRef.

Reply:
yes, will update